src/fen.rs

```diff
@@ -99,14 +101,23 @@ fn row(input: &str) -> IResult<&str, Vec<SquareBuilder>> {
     Ok((input, row))
 }
 
-// For now, this is just whose turn it is.
-fn extra_state(input: &str) -> IResult<&str, Player> {
+fn current_player(input: &str) -> IResult<&str, Player> {
     alt((
         value(Player::White, tag("w")),
         value(Player::Black, tag("b")),
     ))(input)
 }
 
+fn en_passant_pos(input: &str) -> IResult<&str, Option<Pos>> {
+    alt((value(None, tag("-")), map(pos, Some)))(input)
```

Comment:
in fen, there are two options to parse here:

`-` => no en passant,
`e3` (or some valid move) => the en passant target square (i.e., the square where the pawn who captures en passant goes to)